.github/workflows/lint-check.yml

```diff
@@ -0,0 +1,48 @@
+name: Static Analysis
+
+on:
+  pull_request:
+  push:
+
```

Comment:
This deserves comment, since it is different from `build_cmake.yml`, which runs on PRs and pushes to the main/master branch. This check runs on *all* pushes.

Personally, I think it's very useful to run both the build-and-test jobs and the lint-check jobs on every push, so I get immediate feedback on whatever I've done wrong. I too often forget, or am otherwise unable, to run all of these checks locally. I think it's a sane default to run all the checks on all pushes.

On the other hand, running these checks isn't free, and it makes sense not to waste CI minutes indiscriminately. I have blown through all of my free CI minutes for the month doing this. It might be right to restrict these checks to PRs and pushes to main/master, as in `build_cmake.yml`.